extensions/tiled-assets/json-schema/schema.json

```diff
@@ -3,16 +3,49 @@
   "$id": "stac-extension-tiled-assets.json#",
   "title": "Tiled Assets Extension",
   "description": "Tiled Assets Extension to a STAC Item",
-  "allOf": [
+  "oneOf": [
     {
-      "$ref": "../../../item-spec/json-schema/item.json#/definitions/core"
-    },
-    {
-      "$ref": "#/definitions/tiles"
+      "allOf": [
+        {
+          "$ref": "../../../item-spec/json-schema/item.json#/definitions/core"
+        },
+        {
+          "$ref": "#/definitions/item-tiles"
+        }
+      ]
+    }, {
+      "allOf": [
+        {
+          "$ref": "../../../catalog-spec/json-schema/catalog.json"
+        },
+        {
+          "$ref": "#/definitions/catalog-collection-tiles"
+        }
+      ]
+    }, {
+      "allOf": [
+        {
+          "$ref": "../../../collection-spec/json-schema/collection.json"
+        },
+        {
+          "$ref": "#/definitions/catalog-collection-tiles"
+        }
+      ]
     }
   ],
   "definitions": {
-    "tiles": {
+    "catalog-collection-tiles": {
+      "title": "Tile matrix additions to STAC Collections and Catalogs",
+      "description": "Allows the definition of tile matrix sets on a Collection or Catalog.",
+      "type": "object",
+      "required": [],
```

Comment:
I think "tiles:tile_matrix_set" should be required here. Otherwise validation would succeed when nothing is specified, which seems wrong is you specify that a collection/catalogs supports this extension but no field is available.

By the way, I just realized nothing in the extension is actually required. What is the minimum set a user needs to specify? That should be required then....

Reply:
> I think "tiles:tile_matrix_set" should be required here. Otherwise validation would succeed when nothing is specified, which seems wrong is you specify that a collection/catalogs supports this extension but no field is available.

I also wanted to allow the definition of the tile matrix sets on the `Collection` or `Catalog` level, similarly to the `eo:bands`. Does this make sense?

> By the way, I just realized nothing in the extension is actually required. What is the minimum set a user needs to specify? That should be required then....

Hm, I think this would be the `asset_templates`, as without it it would be plain `Items` instead. I'll update the schema.

Reply:
Okay, I think you were right. The extensions to the `Catalog` and `Collection` mean that the only property `tiles:tile_matrix_set` should be required. I changed that.

I also made `asset_templates` mandatory for items.

I think this can be resolved now, what do you think @m-mohr?

Reply:
The schema looks good, but the markdown file / README doesn't reflect the required fields as far as I can see...

Reply:
Thats right. I've added those now.